probes/requiresPRsToChangeCode/def.yml

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
```

Comment:
This is a new probe that was missing from the previous PR with the Branch Protection probes.